internal/controller/protectedvolumereplicationgrouplist_controller_test.go

```diff
@@ -91,11 +92,64 @@ func protectedVrgListExpectIncludeOnly(protectedVrgList *ramen.ProtectedVolumeRe
 	Expect(protectedVrgList.Status.Items).To(ConsistOf(vrgsExpected))
 }
 
+func matchVRG(expected, got ramen.VolumeReplicationGroup) (found, valid bool, msg string) {
+	if expected.Namespace != got.Namespace || expected.Name != got.Name {
+		return false, false, fmt.Sprintf("VRG %s/%s does not match expected: "+
+			"got: %s/%s",
+			expected.Namespace, expected.Name,
+			got.Namespace, got.Name)
+	}
+
+	if expected.Status.ObservedGeneration != got.Status.ObservedGeneration ||
+		expected.Status.State != got.Status.State {
+		return true, false, fmt.Sprintf("VRG %s/%s in pvrgl has mismatched status, expected: "+
+			"Generation: %d, state: %s, "+
+			" got: Generation: %d, state: %s",
+			expected.Namespace, expected.Name,
+			expected.Status.ObservedGeneration, expected.Status.State,
+			got.Status.ObservedGeneration, got.Status.State)
+	}
+
+	return true, true, ""
+}
+
 func protectedVrgListExpectInclude(protectedVrgList *ramen.ProtectedVolumeReplicationGroupList,
 	vrgsExpected []ramen.VolumeReplicationGroup,
 ) {
-	vrgsStatusStateUpdate(protectedVrgList.Status.Items, vrgsExpected)
-	Expect(protectedVrgList.Status.Items).To(ContainElements(vrgsExpected))
+	expectedList := make([]string, 0, len(vrgsExpected))
+	gotList := make([]string, 0, len(protectedVrgList.Status.Items))
+
+	for _, vrgExpected := range vrgsExpected {
+		expectedList = append(expectedList, fmt.Sprintf("%s/%s", vrgExpected.Namespace, vrgExpected.Name))
```

Comment:
Using "vrg" like in the next loop will be better, this is a temporary variable with tiny scope. But since this is only for logging, maybe add a helper like vrgNames() to created the list of names, and call it only if the test fails?